### PR TITLE
Adds commands for creating geometry using alternate methods

### PR DESCRIPTION
**Pull request by tanushree04**
### Pull Request Description

- Updates default method to create building geometry  in create project folder to urban_geometry_creation_zoning.
- Adds optional cli subcommand to specify an alternate method to create building geometry :
   - To create building geometry and add space types using create bar method.
   - To create building geometry and add space types using floorspace.js.

- Adds relevant mapper files, workflow.osw, feature file to example_files folder
- Adds tests to check changes to default project folder and new alternate geometry methods.

### Checklist (Delete lines that don't apply)

- [x] Unit tests have been added or updated
- [x] All ci tests pass (green)
- [x] An [ISSUE](https://github.com/urbanopt/urbanopt-cli/issues) has been created that this is addressing. Issues will get added to the Change Log when the change_log.rb script is run
- [x] This branch is up-to-date with develop

### CODE REVIEW

**Comment by tanushree04** [comment]
The alternate geometry tests would need files to be added to the master branch

**Comment by vtnate** [comment]
@tanushree04 I merged develop into this branch. Be sure you pull to get the updates. Right now it's failing because simulations didn't run because: `Could not set '__SKIP__' to 'false' for measure 'urban_geometry_creation' (RuntimeError)`

File: lib/uo_cli.rb [hunk 1]
```diff
@@ -264,6 +275,18 @@ def self.create_scenario_csv_file(feature_id)
     # +existing_scenario_file+:: _string_ - Name of existing ScenarioFile
     def self.create_reopt_scenario_file(existing_scenario_file)
       existing_path, existing_name = File.split(File.absolute_path(existing_scenario_file))
+      
+      # make reopt folder
+      Dir.mkdir File.join(existing_path, "reopt")
```
Comment (tanushree04) [inline lib/uo_cli.rb:280]: Makes reopt folder for reopt subcommand

File: lib/uo_cli.rb [hunk 2]
```diff
@@ -292,10 +315,85 @@ def self.create_project_folder(dir_name, empty_folder = false, overwrite_project
...
+              # copy feature file
+              FileUtils.cp(File.join(path_item, "example_project.json"), dir_name)
+
+              # copy osm 
+              FileUtils.cp(File.join(path_item, "osm_building/7.osm"), File.join(dir_name, "osm_building"))
+              FileUtils.cp(File.join(path_item, "osm_building/8.osm"), File.join(dir_name, "osm_building"))
+              FileUtils.cp(File.join(path_item, "osm_building/9.osm"), File.join(dir_name, "osm_building"))
+
+            
+              if @opthash.subopts[:create_bar] == false
```
Comment (tanushree04) [inline lib/uo_cli.rb:357]: copying files for default geometry method

File: lib/uo_cli.rb [hunk 3]
```diff
@@ -292,10 +315,85 @@ def self.create_project_folder(dir_name, empty_folder = false, overwrite_project
...
+
+                # copy the mappers
+                FileUtils.cp(File.join(path_item, "mappers/Baseline.rb"), File.join(dir_name, "mappers"))
+                FileUtils.cp(File.join(path_item, "mappers/HighEfficiency.rb"), File.join(dir_name, "mappers"))
+                FileUtils.cp(File.join(path_item, "mappers/ThermalStorage.rb"), File.join(dir_name, "mappers"))
+
+                # copy osw file
+                FileUtils.cp(File.join(path_item, "mappers/base_workflow.osw"), File.join(dir_name, "mappers"))
+              
+              elsif @opthash.subopts[:create_bar] == true
```
Comment (tanushree04) [inline lib/uo_cli.rb:367]: copying files for create bar geometry method

File: lib/uo_cli.rb [hunk 4]
```diff
@@ -292,10 +315,85 @@ def self.create_project_folder(dir_name, empty_folder = false, overwrite_project
...
+                # copy the mappers
+                FileUtils.cp(File.join(path_item, "mappers/CreateBar.rb"), File.join(dir_name, "mappers"))
+                FileUtils.cp(File.join(path_item, "mappers/HighEfficiencyCreateBar.rb"), File.join(dir_name, "mappers"))
+
+                # copy osw file
+                FileUtils.cp(File.join(path_item, "mappers/createbar_workflow.osw"), File.join(dir_name, "mappers"))
+              
+              end
+            
+            elsif @opthash.subopts[:floorspace] == true
```
Comment (tanushree04) [inline lib/uo_cli.rb:378]: copying files for floorspace geomtry method

File: spec/uo_cli_spec.rb [hunk 5]
```diff
@@ -42,6 +42,19 @@ def delete_directory_or_file(dir_or_file)
...
+    it 'creates an example project directory when create bar geometry method specified' do
+      system("#{call_cli} create --project-folder #{test_directory} --create-bar")
+      expect(File.exist?(File.join(test_directory, 'mappers/CreateBar.rb'))).to be true
+      expect(File.exist?(File.join(test_directory, 'mappers/createbar_workflow.osw'))).to be true
+    end
+
+    it 'creates an example project directory when floorspace method specified' do
+      system("#{call_cli} create --project-folder #{test_directory} --floorspace")
+      expect(File.exist?(File.join(test_directory, 'mappers/Floorspace.rb'))).to be true
+      expect(File.exist?(File.join(test_directory, 'example_floorspace_project.json')))
```
Comment (tanushree04) [inline spec/uo_cli_spec.rb:57]: tests for creating project for different geometry methods

File: spec/uo_cli_spec.rb [hunk 6]
```diff
@@ -108,19 +123,49 @@ def delete_directory_or_file(dir_or_file)
...
+      system("cp #{File.join('example_files', 'osm_building', '7_floorspace.json')} #{File.join(test_directory, 'osm_building', '7_floorspace.json')}")
+      system("cp #{File.join('example_files', 'osm_building', '7_floorspace.osm')} #{File.join(test_directory, 'osm_building', '7_floorspace.osm')}")
+      # Copy feature file
+      system("cp #{File.join('example_files', 'example_floorspace_project.json')} #{File.join(test_directory, 'example_floorspace_project.json')}")
+      # Copy scenario file with 2 buildings in it
+      system("cp #{File.join('spec', 'spec_files', 'two_building_floorspace.csv')} #{File.join(test_directory, 'two_building_floorspace.csv')}")
+      system("#{call_cli} run --scenario #{File.join(test_directory, 'two_building_floorspace.csv')} --feature #{File.join('../example_files/example_floorspace_project.json')}")
+      expect(File.exist?(File.join(test_directory, 'run', 'two_building_floorspace', '1', 'finished.job'))).to be true
+      expect(File.exist?(File.join(test_directory, 'run', 'two_building_floorspace', '7', 'finished.job'))).to be true
     end
```
Comment (tanushree04) [inline spec/uo_cli_spec.rb:162]: tests for running projects for different geometry methods

File: spec/uo_cli_spec.rb [hunk 7]
```diff
@@ -149,13 +194,13 @@ def delete_directory_or_file(dir_or_file)
...
 
     it 'post-processes a scenario' do
-      filename = File.join(test_directory, 'run', 'two_building_scenario', 'default_scenario_report.csv')
-      db_filename = File.join(test_directory, 'run', 'two_building_scenario', 'default_scenario_report.db')
+      filename = File.join(test_directory, 'run', 'one_building_scenario', 'default_scenario_report.csv')
```
Comment (kflemin) [inline spec/uo_cli_spec.rb:197]: shouldn't we have at least 2 buildings in the scenario to ensure the aggregations are working?

**Review by kflemin — APPROVED** [review]

File: spec/uo_cli_spec.rb (hunk 7 above)
Comment (tanushree04) [reply spec/uo_cli_spec.rb:197]: Yeah, it would make more sense for feature visualization too to have two features. It was reduced to 1 only because running the rake tasks was taking too long, since it was running 2 features for 3 geometry types

File: Gemfile [hunk 8]
```diff
@@ -60,6 +60,6 @@ else
   gem 'urbanopt-reopt', github: 'URBANopt/urbanopt-reopt-gem', branch: 'develop'
 end
 
-gem 'openstudio-load-flexibility-measures', github: 'NREL/openstudio-load-flexibility-measures-gem', branch: 'v0.1.3'
+gem 'openstudio-load-flexibility-measures', github: 'NREL/openstudio-load-flexibility-measures-gem', branch: 'master'
```
Comment (vtnate) [inline Gemfile:64]: I would prefer this to be pinned to the release instead of to `master`, so there is a bit more stability. Can you revert this change? As of right now v0.1.3 == master

File: example_files/Gemfile [hunk 9]
```diff
@@ -45,7 +45,7 @@ end
 #if allow_local && File.exist?('../openstudio-load-flexibility-measures-gem')
 #  gem 'openstudio-load-flexibility-measures', path: '../openstudio-load-flexibility-measures-gem'
 #elsif allow_local
-  gem 'openstudio-load-flexibility-measures', github: 'NREL/openstudio-load-flexibility-measures-gem', branch: 'v0.1.3'
+  gem 'openstudio-load-flexibility-measures', github: 'NREL/openstudio-load-flexibility-measures-gem', branch: 'master'
```
Comment (vtnate) [inline example_files/Gemfile:48]: Same here, please use the released version

File: spec/uo_cli_spec.rb [hunk 10]
```diff
@@ -1,7 +1,7 @@
 
 RSpec.describe URBANopt::CLI do
   test_directory = File.join('spec', 'test_directory')
-  test_scenario = File.join(test_directory, 'two_building_scenario.csv')
+  test_scenario = File.join(test_directory, 'one_building_scenario.csv')
```
Comment (vtnate) [inline spec/uo_cli_spec.rb:4]: Since your tests passed, can we put this back to the 2 building scenario?

**Review by vtnate — CHANGES_REQUESTED** [review]

**Comment by vtnate** [comment]
This is great! I have a few minor requests, otherwise I'm good with it!

**Review by vtnate — APPROVED** [review]
Looks good!
